Modules/zlibmodule.c

```diff
@@ -1876,6 +1886,44 @@ zlib_adler32_impl(PyObject *module, Py_buffer *data, unsigned int value)
     return PyLong_FromUnsignedLong(value & 0xffffffffU);
 }
 
+/*[clinic input]
+zlib.adler32_combine -> unsigned_int
+
+    adler1: unsigned_int(bitwise=True)
+        Adler-32 check value for sequence A
+
+    adler2: unsigned_int(bitwise=True)
+        Adler-32 check value for sequence B
+
+    blen: object
+        Length of sequence B
+    /
+
+Combine two Adler-32 check values into one.
+
+Given an Adler-32 check value 'adler1' of a sequence A and an Adler-32 check
+value 'adler2' of a sequence B of length 'blen', the returned checksum
+is the Adler-32 check value of A and B concatenated.
+[clinic start generated code]*/
+
+static unsigned int
+zlib_adler32_combine_impl(PyObject *module, unsigned int adler1,
+                          unsigned int adler2, PyObject *blen)
```

Comment:
zlib uses `len2` instead of `blen` here, which I think makes more sense (and agrees with the .rst doc)